crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -40,31 +40,54 @@ def supported?
 
     def knife_files
       @data.join("knife", "databags", "barclamps").rmtree
+      knife_path = @data.join("knife")
 
-      crowbar_databags_path = @data.join("knife", "databags", "crowbar")
+      crowbar_databags_path = knife_path.join("databags", "crowbar")
       crowbar_databags_path.children.each do |file|
-        if file == "bc-template-nova_dashboard.json"
-          crowbar_databags_path.join(file).rename(
-            crowbar_databags_path.join(file.sub!("nova_dashboard", "horizon"))
-          )
+        file_path = crowbar_databags_path.join(file)
 
-          file_content = File.read(crowbar_databags_path.join(file))
-          file_content.gsub!("nova_dashboard", "horizon")
-          File.open(crowbar_databags_path.join(file), "w") { |content| content.puts file_content }
+        if file == "bc-template-nova_dashboard.json"
+          new_file = filename_replace(file_path, "nova_dashboard", "horizon")
+          filecontent_replace(new_file, "nova_dashboard", "horizon")
+          file_path = new_file
         end
 
-        next unless file.to_s.match("bc-.*.json")
-        new_file = file.sub("bc-", "")
-        crowbar_databags_path.join(file).rename(new_file)
+        next unless file_path.to_s =~ "bc-.*.json"
+        file_path = filename_replace(file_path, "bc-", "")
+        filecontent_replace(file_path, "bc-", "")
+      end
 
-        file_content = crowbar_databags_path.join(new_file).read
-        file_content.gsub!("bc-", "")
-        File.open(crowbar_databags_path.join(new_file), "w") { |content| content.puts file_content }
+      barclamps_databag_path = knife_path.join("barclamps")
```

Comment:
That path (..../knife/barclamps) doesn't exist in the data dump AFAICS. I guess you meant (.../knife/databags/barclamps), bugt we're removing that directory on line 42. So something is wrong here. I don't think we need those files for the upgrade.